### ISSUE TITLE
False positive: `unexpected-keyword-argument` when passing `dtype` to `numpy.concatenate()`

### ISSUE THREAD

**Issue by SnoopJ** [pylint-dev/pylint#10548]
### Bug description

`pylint` reports an unexpected keyword argument when passing `dtype` to [`numpy.concatenate()`](https://numpy.org/doc/2.0/reference/generated/numpy.concatenate.html)

### Configuration

N/A

### Command used

```shell
python3 -m pylint -E pylint_wat.py
```

### Pylint output

```python
$ python3 -m pylint -E pylint_wat.py 
************* Module pylint_wat
pylint_wat.py:5:9: E1123: Unexpected keyword argument 'dtype' in function call (unexpected-keyword-arg)
```

### Expected behavior

There should be no error. The program is correct, and in fact `dtype` can _only_ be passed by keyword.

```
$ python3 pylint_wat.py 
[[ 1.  2.  3.  7.  8.  9.]
 [ 4.  5.  6. 10. 11. 12.]]
```

### Pylint version

```shell
$ python3 -m pylint --version
pylint 3.3.8
astroid 3.3.11
Python 3.9.16 (main, Mar 15 2023, 15:36:41) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 22.04
Python 3.9.16

### Additional dependencies

```python
False positive occurs with `numpy` 1.26 as well as 2.0.2
```
